source/core/slang-type-text-util.cpp

```diff
@@ -36,13 +37,43 @@ struct ScalarTypeInfo
     UnownedStringSlice text;
 };
 
-static const ScalarTypeInfo s_scalarTypeInfo[] =
+static const ScalarTypeInfo s_scalarTypeInfos[] =
 {
     #define SLANG_SCALAR_TYPE_INFO(value, text) \
             { slang::TypeReflection::ScalarType::value, UnownedStringSlice::fromLiteral(#text) },
     SLANG_SCALAR_TYPES(SLANG_SCALAR_TYPE_INFO)
 };
 
+struct CompileTargetInfo
+{
+    SlangCompileTarget target;          ///< The target
+    const char* extensions;             ///< Comma delimited list of extensions associated with the target
+    const char* names;                  ///< Comma delimited list of names associated with the target. NOTE! First name is taken as the normal display name.
+};
+
+static const CompileTargetInfo s_compileTargetInfos[] = 
+{
+    { SLANG_TARGET_UNKNOWN, "",                                                 "unknown"},
+    { SLANG_TARGET_NONE,    "",                                                 "none"},
+    { SLANG_HLSL,           "hlsl,fx",                                          "hlsl"},
+    { SLANG_DXBC,           "dxbc",                                             "dxbc"},
+    { SLANG_DXBC_ASM,       "dxbc.asm",                                         "dxbc-asm,dxbc-assembly" },
+    { SLANG_DXIL,           "dxil",                                             "dxil" },
+    { SLANG_DXIL_ASM,       "dxil.asm",                                         "dxil-asm,dxil-assembly" },
+    { SLANG_GLSL,           "glsl,vert,frag,geom,tesc,tese,comp",               "glsl" },
+    { SLANG_GLSL_VULKAN,    "",                                                 "glsl-vulkan" },
+    { SLANG_GLSL_VULKAN_ONE_DESC, "",                                           "glsl-vulkan-one-desc" },
```

Comment:
Unrelated, but these should probably be considered fully deprecated and ready for removal. No current client code is passing these in, and the `*_ONE_DESC` variant was specifically created for the Slang research effort.

If/when we start to distinguish Vulkan GLSL from OpenGL GLSL, we will need to invent a new way to do it.